Fails to compile for Linux & Android platforms

Fixed in PR #23 

Build error reported when addign the plug in to a project with a Linux build will be:
```

[11:02:08] :	 [Step 1/1] Building 7 actions with 64 processes...
[11:02:09] :	 [Step 1/1]   ** For VrProject4_26-Linux-Development
[11:02:09] :	 [Step 1/1]   BuildSettings.cpp
[11:02:18] :	 [Step 1/1]   Module.Gauntlet.gen.cpp
[11:02:19] :	 [Step 1/1]   Module.DaedalicTestAutomationPlugin.gen.cpp
[11:02:19] :	 [Step 1/1]   Module.Gauntlet.cpp
[11:02:21] :	 [Step 1/1]   Module.DaedalicTestAutomationPlugin.cpp
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent/work/85a46caaa9dd1588/VrProject4_26/Intermediate/Build/Linux/B4D820EA/VrProject4_26/Development/DaedalicTestAutomationPlugin/Module.DaedalicTestAutomationPlugin.cpp:1:
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent\work\85a46caaa9dd1588\VrProject4_26\Intermediate\Build\Linux\B4D820EA\VrProject4_26\Development\Engine\SharedPCH.Engine.ShadowErrors.h:168:
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Engine\Public\EngineSharedPCH.h:86:
[11:02:21] :	 [Step 1/1]   In file included from D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointer.h:113:
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(164,4): error: delete called on non-final 'FDaeTestReportWriterJUnit' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-abstract-non-virtual-dtor]
[11:02:21] :	 [Step 1/1]                           delete Object;
[11:02:21] :	 [Step 1/1]                           ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(116,4): note: in instantiation of member function 'SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterJUnit>::operator()' requested here
[11:02:21] :	 [Step 1/1]                           (*static_cast<DeleterType*>(this))(Object);
[11:02:21] :	 [Step 1/1]                           ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(108,12): note: in instantiation of member function 'SharedPointerInternals::TReferenceControllerWithDeleter<FDaeTestReportWriterJUnit, SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterJUnit> >::DestroyObject' requested here
[11:02:21] :	 [Step 1/1]                   explicit TReferenceControllerWithDeleter(ObjectType* InObject, DeleterType&& Deleter)
[11:02:21] :	 [Step 1/1]                            ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(172,14): note: in instantiation of member function 'SharedPointerInternals::TReferenceControllerWithDeleter<FDaeTestReportWriterJUnit, SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterJUnit> >::TReferenceControllerWithDeleter' requested here
[11:02:21] :	 [Step 1/1]                   return new TReferenceControllerWithDeleter<ObjectType, DefaultDeleter<ObjectType>>(Object, DefaultDeleter<ObjectType>());
[11:02:21] :	 [Step 1/1]                              ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(204,27): note: in instantiation of function template specialization 'SharedPointerInternals::NewDefaultReferenceController<FDaeTestReportWriterJUnit>' requested here
[11:02:21] :	 [Step 1/1]                           , ReferenceController( NewDefaultReferenceController( InObject ) )
[11:02:21] :	 [Step 1/1]                                                  ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointer.h(1817,9): note: in instantiation of member function 'SharedPointerInternals::FRawPtrProxy<FDaeTestReportWriterJUnit>::FRawPtrProxy' requested here
[11:02:21] :	 [Step 1/1]           return SharedPointerInternals::FRawPtrProxy< ObjectType >( InObject );
[11:02:21] :	 [Step 1/1]                  ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\VrProject4_26\Plugins\DaedalicTestAutomationPlugin\Source\DaedalicTestAutomationPlugin\Private\DaeTestActor.cpp(107,23): note: in instantiation of function template specialization 'MakeShareable<FDaeTestReportWriterJUnit>' requested here
[11:02:21] :	 [Step 1/1]       ReportWriters.Add(MakeShareable(new FDaeTestReportWriterJUnit()));
[11:02:21] :	 [Step 1/1]                         ^
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent/work/85a46caaa9dd1588/VrProject4_26/Intermediate/Build/Linux/B4D820EA/VrProject4_26/Development/DaedalicTestAutomationPlugin/Module.DaedalicTestAutomationPlugin.cpp:1:
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent\work\85a46caaa9dd1588\VrProject4_26\Intermediate\Build\Linux\B4D820EA\VrProject4_26\Development\Engine\SharedPCH.Engine.ShadowErrors.h:168:
[11:02:21] :	 [Step 1/1]   In file included from D:/BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Engine\Public\EngineSharedPCH.h:86:
[11:02:21] :	 [Step 1/1]   In file included from D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointer.h:113:
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(164,4): error: delete called on non-final 'FDaeTestReportWriterPerformance' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-abstract-non-virtual-dtor]
[11:02:21] :	 [Step 1/1]                           delete Object;
[11:02:21] :	 [Step 1/1]                           ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(116,4): note: in instantiation of member function 'SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterPerformance>::operator()' requested here
[11:02:21] :	 [Step 1/1]                           (*static_cast<DeleterType*>(this))(Object);
[11:02:21] :	 [Step 1/1]                           ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(108,12): note: in instantiation of member function 'SharedPointerInternals::TReferenceControllerWithDeleter<FDaeTestReportWriterPerformance, SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterPerformance> >::DestroyObject' requested here
[11:02:21] :	 [Step 1/1]                   explicit TReferenceControllerWithDeleter(ObjectType* InObject, DeleterType&& Deleter)
[11:02:21] :	 [Step 1/1]                            ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(172,14): note: in instantiation of member function 'SharedPointerInternals::TReferenceControllerWithDeleter<FDaeTestReportWriterPerformance, SharedPointerInternals::DefaultDeleter<FDaeTestReportWriterPerformance> >::TReferenceControllerWithDeleter' requested here
[11:02:21] :	 [Step 1/1]                   return new TReferenceControllerWithDeleter<ObjectType, DefaultDeleter<ObjectType>>(Object, DefaultDeleter<ObjectType>());
[11:02:21] :	 [Step 1/1]                              ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointerInternals.h(204,27): note: in instantiation of function template specialization 'SharedPointerInternals::NewDefaultReferenceController<FDaeTestReportWriterPerformance>' requested here
[11:02:21] :	 [Step 1/1]                           , ReferenceController( NewDefaultReferenceController( InObject ) )
[11:02:21] :	 [Step 1/1]                                                  ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\Engine\Source\Runtime\Core\Public\Templates\SharedPointer.h(1817,9): note: in instantiation of member function 'SharedPointerInternals::FRawPtrProxy<FDaeTestReportWriterPerformance>::FRawPtrProxy' requested here
[11:02:21] :	 [Step 1/1]           return SharedPointerInternals::FRawPtrProxy< ObjectType >( InObject );
[11:02:21] :	 [Step 1/1]                  ^
[11:02:21] :	 [Step 1/1]   D:\BuildAgent\work\85a46caaa9dd1588\VrProject4_26\Plugins\DaedalicTestAutomationPlugin\Source\DaedalicTestAutomationPlugin\Private\DaeTestPerformanceBudgetActor.cpp(280,23): note: in instantiation of function template specialization 'MakeShareable<FDaeTestReportWriterPerformance>' requested here
[11:02:21] :	 [Step 1/1]       ReportWriters.Add(MakeShareable(new FDaeTestReportWriterPerformance()));
[11:02:21] :	 [Step 1/1]                         ^
[11:02:21] :	 [Step 1/1]   2 errors generated.
[11:02:21] :	 [Step 1/1] BUILD FAILED:  failed, retries not enabled:
[11:02:21] :	 [Step 1/1] AutomationTool exiting with ExitCode=1 (Error_Unknown)
[11:02:21] :	 [Step 1/1] BUILD FAILED
[11:02:21] :	 [Step 1/1] 
```

--- a/DaedalicTestAutomationPlugin/Source/DaedalicTestAutomationPlugin/Public/DaeTestReportWriter.h
+++ b/DaedalicTestAutomationPlugin/Source/DaedalicTestAutomationPlugin/Public/DaeTestReportWriter.h
@@ -14,6 +14,9 @@ public:
     virtual void WriteReport(const TArray<FDaeTestSuiteResult>& TestSuites,
                              const FString& ReportPath) const = 0;
 
+    /** virtual destructor, required for Linux/Android builds */
+    virtual ~FDaeTestReportWriter() {}
+
 protected:
     /** Gets the total number of tests among the passed test suites. */
     int32 NumTotalTests(const TArray<FDaeTestSuiteResult>& TestSuites) const;

